types/src/block/block_header/block_header_v2.rs

```diff
@@ -81,10 +83,12 @@ pub struct BlockHeaderV2 {
     pub(super) proposer: PublicKey,
     /// The gas price of the era
     pub(super) current_gas_price: u8,
+    /// The most recent switch block hash.
+    pub(super) last_switch_block_hash: Digest,
```

Comment:
This should be BlockHash, not Digest. 

Also, we have to consider the genesis block (on a new network)...either this is an Option<BlockHash> or the Genesis immediate switch block would cite itself and the logic would deal w/ the special case of self reference.

Reply:
So in the case that its the genesis era, its a special case `BlockHash::default()`

Reply:
ok